crates/worker/src/purgatory.rs

```diff
@@ -116,6 +111,63 @@ impl Inner {
             .push((id, to_purge_timestampe));
     }
 
+    async fn global_state_purge(db_client: DynamoDBClient, s3_client: S3Client) {
+        const SYNC_FROM_OFFSET: Option<Duration> = PURGE_INTERVAL.checked_mul(6);
+
+        let mut global_state = GlobalState::new(db_client.clone(), s3_client.clone());
+        let sync_from = Utc::now() - SYNC_FROM_OFFSET.unwrap();
+
+        loop {
+            if global_state.sync(&sync_from).await.is_err() {
+                break;
+            }
+            if global_state.items.is_empty() {
+                break;
+            }
+
+            let items: Vec<Item> = global_state.items.drain(..).collect();
+            for item in items {
+                Inner::purge_item(&db_client, &s3_client, &item.id, &item.status).await;
+            }
+        }
+    }
+
+    pub async fn purge_item(
+        db_client: &DynamoDBClient,
+        s3_client: &S3Client,
+        id: &Uuid,
+        status: &Status,
+    ) {
```

Comment:
I think we should return `Result<>` here and avoid unwraps later

Reply:
This also related to retriable clients, done in the latest pr